pkg/pillar/cmd/zedagent/handlecipherconfig.go

```diff
@@ -133,134 +59,196 @@ func parseCipherContextConfig(getconfigCtx *getconfigContext,
 			log.Debugf("parseCipherContextConfig ignoring empty\n")
 			continue
 		}
-		cipherContext := new(types.CipherContext)
-		cipherContext.ID = cfgCipherContext.GetContextId()
-		cipherContext.HashScheme = cfgCipherContext.GetHashScheme()
-		cipherContext.KeyExchangeScheme = cfgCipherContext.GetKeyExchangeScheme()
-		cipherContext.EncryptionScheme = cfgCipherContext.GetEncryptionScheme()
-		cipherContext.DeviceCertHash = cfgCipherContext.GetDeviceCertHash()
-		cipherContext.ControllerCertHash = cfgCipherContext.GetControllerCertHash()
-		log.Debugf("parseCipherContextConfig publishing %v\n", cipherContext)
-		publishCipherContextConfig(getconfigCtx, cipherContext)
+		cipherCtx := types.CipherContext{}
+		cipherCtx.ContextID = cfgCipherContext.GetContextId()
+		cipherCtx.HashScheme = cfgCipherContext.GetHashScheme()
+		cipherCtx.KeyExchangeScheme = cfgCipherContext.GetKeyExchangeScheme()
+		cipherCtx.EncryptionScheme = cfgCipherContext.GetEncryptionScheme()
+		cipherCtx.DeviceCertHash = cfgCipherContext.GetDeviceCertHash()
+		cipherCtx.ControllerCertHash = cfgCipherContext.GetControllerCertHash()
```

Comment:
Why not intialize the fields inside the {} ?

Reply:
sure, will do.